src/briefcase/commands/dev.py

```diff
@@ -133,18 +139,19 @@ def __call__(
         # installed. If a dependency update has been manually requested,
         # do it regardless.
         dist_info_path = self.app_module_path(app).parent / '{app.module_name}.dist-info'.format(app=app)
-        if update_dependencies or not dist_info_path.exists():
+        if update_dependencies or no_run or not dist_info_path.exists():
```

Comment:
It took me a moment to work out why `no_run` was being mentioned here. Rather than integrating the no_run check here, it might be better to do a once-off check modifying `update_dependencies` if no_run is specified. That simplifies the logic on this line, and if we ever need to add some other check against update_dependencies, we won't accidentally forget the impact of no-run.